classes/class-wc-connect-shipping-label.php

```diff
@@ -132,13 +132,37 @@ protected function get_states_map() {
 			return $result;
 		}
 
-		public function meta_box( $post ) {
-			global $theorder;
+		public function should_show_meta_box() {
+			$order = wc_get_order();
+
+			if ( ! $order ) {
+				return false;
+			}
+
+			// TODO: return true if the order has already label meta-data
 
-			if ( ! is_object( $theorder ) ) {
-				$theorder = wc_get_order( $post->ID );
+			$base_location = wc_get_base_location();
```

Comment:
In the shipping method instance we only require the postcode. Plus, since labels and shipping rates are independent, it can be that no shipping methods are configured. The only "reliable" way of getting the warehouse's country is reading the `Base Location` of WooCommerce.